Add initial doc changes related to tracing provider and support for z… (#2884)
istio/istio.io: content/docs/tasks/telemetry/distributed-tracing/overview.md

> After completing this task, you should understand all of the assumptions about your application and how to have it participate in tracing, regardless of what language/framework/platform you use to build your application. The [Bookinfo](/docs/examples/bookinfo/) sample is used as the example application for this task.

Reviewer: What do you mean with "assumptions about your application"? Who is doing the assuming? Are these Istio assumptions? Pilot's? The reader's? Please mind the 80 character line length. Fixing grammar issues:
```suggestion
After completing this task, you understand all of the assumptions about your application. You understand how to have your application participate in tracing, regardless of the language, framework, or platform you use to build your application. This task uses thee [Bookinfo](/docs/examples/bookinfo/) sample as the example application.
```

Author: Agree it is not clear - wasn't my text (just reorganised) - will remove that part.
